pandas/core/groupby/ops.py

```diff
@@ -241,6 +242,11 @@ def apply(self, f: F, data: FrameOrSeries, axis: int = 0):
     @cache_readonly
     def indices(self):
         """ dict {group name -> group indices} """
+        if len(self.groupings) == 1 and isinstance(
```

Comment:
woa, this is so special casing here. why is this added?

Reply:
See the explanation in the top post

Reply:
thx

Reply:
I read it and don't buy it. Why are we supporting this case at all? I would rather simply fix this breaking or not.

Reply:
You mean not showing unused categories for indices?

Reply:
i c we did thave this special case before, but *onl* was for len(self.groupins) == 1. its the special isinstance i object.